control-plane/api/v1alpha1/mesh_types.go

```diff
@@ -194,11 +206,12 @@ func (in *Mesh) MatchesConsul(candidate capi.ConfigEntry) bool {
 	return cmp.Equal(in.ToConsul(""), configEntry, cmpopts.IgnoreFields(capi.MeshConfigEntry{}, "Partition", "Namespace", "Meta", "ModifyIndex", "CreateIndex"), cmpopts.IgnoreUnexported(), cmpopts.EquateEmpty())
 }
 
-func (in *Mesh) Validate(_ common.ConsulMeta) error {
+func (in *Mesh) Validate(consulMeta common.ConsulMeta) error {
 	var errs field.ErrorList
 	path := field.NewPath("spec")
 
 	errs = append(errs, in.Spec.TLS.validate(path.Child("tls"))...)
+	errs = append(errs, in.Spec.Peering.validate(path.Child("peering"), consulMeta.PartitionsEnabled, consulMeta.Partition)...)
```

Comment:
There is an additional validation in Consul that mesh CRs are always in the default namespace, but that always seems to be implied here on creation. I don't see where namespace is transferred in `ToConsul`.